### PR TITLE
Eosio build darwin: extensive validation

### PR DESCRIPTION
**Pull request by pacificcode**
Added check for cmake cache file in eos/ doc root.
Added extensive command validation error checking in darwin build.

### CODE REVIEW

File: eosio_build.sh [hunk 1]
```diff
@@ -35,6 +35,10 @@
 		printf "\\n\\tPlease cd into directory %s to run this script.\\n \\tExiting now.\\n\\n" "${CWD}"
 		exit 1
 	fi
+	if [ -f "${PWD}/CMakeCache.txt" ]; then
+		printf "\\n\\tPlease remove file %s/CMakeCache.txt before building EOS.IO.\\n \\tExiting now.\\n\\n" "${PWD}"
+		exit 1
+	fi
```
Comment (heifner) [inline eosio_build.sh]: I thought people used the eosio_build.sh script to rebuild their eosio.io after an update. This prevents eosio_build.sh from working for an existing checkout with a previous build.

File: scripts/eosio_build_darwin.sh [hunk 2]
```diff
@@ -211,170 +214,302 @@
...
+		printf "\\tBoost 1.66.0 found at /usr/local.\\n"
 	fi
 
-	printf "\n\tChecking MongoDB C++ driver installation.\n"
-    if [ ! -e /usr/local/lib/libmongocxx-static.a ]; then
-		cd ${TEMP_DIR}
-		brew install --force pkgconfig
-		brew unlink pkgconfig && brew link --force pkgconfig
+	printf "\\n\\tChecking MongoDB C++ driver installation.\\n"
+    if [ ! -e "/usr/local/lib/libmongocxx-static.a" ]; then
```
Comment (heifner) [inline scripts/eosio_build_darwin.sh]: Note the slim branch does not have a mongodb plugin and will likely not have one for June.

File: eosio_build.sh (hunk 1 above)
Comment (pacificcode) [reply eosio_build.sh]: This file is required in directory /path/to/eos/build/
If this file is located in the eos/ directory CMake will write its build files to the eos/ directory and not the build directory which will cause:
cd build/
make make -j"${CPU_CORE}"
to error out i.e.
make: *** No targets specified and no makefile found.  Stop.

File: scripts/eosio_build_darwin.sh (hunk 2 above)
Comment (pacificcode) [reply scripts/eosio_build_darwin.sh]: I have $( git branch | grep \* ) writing the current branch to stdout.
I can add a check in to install mongo when necessary. We need to define what that condition/branch is.

Comment (heifner) [reply scripts/eosio_build_darwin.sh]: It will be master branch soon.

File: eosio_build.sh (hunk 1 above)
Comment (heifner) [reply eosio_build.sh]: Hmm. I build in source, but I guess build script does out of source. ok.

**Pull request merged**